lxd/instance_logs.go

```diff
@@ -238,13 +239,25 @@ func instanceLogGet(d *Daemon, r *http.Request) response.Response {
 		return response.BadRequest(err)
 	}
 
-	if !validLogFileName(file) {
-		return response.BadRequest(fmt.Errorf("log file name %s not valid", file))
+	fileName := filepath.Base(file)
+
+	if !validLogFileName(fileName) {
+		return response.BadRequest(fmt.Errorf("log file name %s not valid", fileName))
+	}
+
+	var filePath string
+	switch filepath.Dir(file) {
+	case ".":
+		filePath = shared.LogPath(project.Instance(projectName, name), fileName)
+	case "exec-output":
+		filePath = filepath.Join(inst.ExecOutputPath(), fileName)
```

Comment:
Can you test what the old behaviour was, but I think it was possible to do an `lxc exec` then stop the instance and then be able to retrieve the output after stopping the instance. 

Now we are storing the exec output underneath the instance's root volume, in order for the files to be accessible even when the instance is off (for all storage pool types) we need to mount the instance's volume first in this case.

Reply:
I have tested with storage pools of type `dir` and `zfs`. When an instance is stopped it works only for an instance with storage type of `dir`.

Reply:
Is this now or before?

Reply:
I've tested and I can confirm that in previous implementation it works perfectly fine when the container is stopped (both getting and deleting the exec-output)